src/plugins/template/tests/functional/op_reference/base_reference_test.hpp

```diff
@@ -57,6 +57,11 @@ ov::Tensor CreateTensor(const ov::element::Type& element_type, const std::vector
 // Create blob with correct input shape (not 1-dimensional). Will be used in tests with dynamic input shapes
 template <class T>
 ov::Tensor CreateTensor(const ov::Shape& shape, const ov::element::Type& element_type, const std::vector<T>& values) {
+    if (element_type == ov::element::string) {
+        ov::Tensor tensor{element_type, shape};
+        std::copy_n(values.data(), shape_size(shape), tensor.data<T>());
+        return tensor;
+    }
```

Comment:
It might break RVO.

Reply:
Any suggestions for particular improvement?
This is a part of the test util corresponding to the existing code for the numeric types below:
https://github.com/openvinotoolkit/openvino/blob/c29bc760b176bce1202dc51261bfcd49eb459f88/src/plugins/template/tests/functional/op_reference/base_reference_test.hpp#L65-L72
Assuming the copy of the test data is done to ensure the life time, 
Is there a constructor of the ov::Tensor making a copy of the data from the std::vector?

Reply:
```cpp
    ov::Tensor tensor{element_type, shape};
    if (element_type == ov::element::string) {    
         std::copy_n(values.data(), shape_size(shape), tensor.data<T>());
    } else {
        size_t size = sizeof(T) * values.size();
        if (tensor.get_byte_size() < size)
            size = tensor.get_byte_size();
        std::memcpy(tensor.data(), values.data(), size);
    }
    return tensor;
```

1. Create tensor.
2. Depends on element type apply action.
3. return created tensor, RVO will be applied (always same object returned).

Reply:
I should be more correct the solution will apply NRVO.